packages/react-sdk/src/components/html.tsx

```diff
@@ -0,0 +1,94 @@
+import {
+  forwardRef,
+  useContext,
+  useEffect,
+  useRef,
+  type ForwardedRef,
+} from "react";
+import { mergeRefs } from "@react-aria/utils";
+import { ReactSdkContext } from "../context";
+
+type Props = {
+  code: string;
+  executeScriptInCanvas: boolean;
+};
+
+type ChildProps = {
+  innerRef: ForwardedRef<HTMLDivElement>;
+  // code can be actually undefined when prop is not provided
+  code?: string;
+};
+
+/**
+ * Scripts are executed when rendered client side
+ */
+const ExecutableHtml = (props: ChildProps) => {
+  const { code, innerRef, ...rest } = props;
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const container = containerRef.current;
+    if (container === null || code === undefined) {
+      return;
+    }
+    // the trick to execute inserted scripts
+    const range = document.createRange();
+    range.setStart(container, 0);
+    const fragment = range.createContextualFragment(code);
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+    container.append(fragment);
+  }, [code]);
+
+  return (
+    <div
+      {...rest}
+      ref={mergeRefs(innerRef, containerRef)}
+      style={{ display: "contents" }}
+    />
+  );
+};
+
+/**
+ * Scripts are executed when rendered server side
+ */
+const InnerHtml = (props: ChildProps) => {
+  const { code, innerRef, ...rest } = props;
+
+  return (
+    <div
+      {...rest}
+      ref={innerRef}
+      style={{ display: "contents" }}
+      dangerouslySetInnerHTML={{ __html: props.code ?? "" }}
```

Comment:
Probably I'm wrong but this will ruin layout

Reply:
I just tested with SSR and this ruins layout

<img width="406" alt="image" src="https://github.com/webstudio-is/webstudio-builder/assets/5077042/1914926f-3774-43bc-be19-3597562f6486">

This should not be used without html sanitizer which will enclose each tag

Reply:
I think it will be perfectly ok to use client-side injection here too, less error-prone.
But probably some scripts depending on load even will not work.